frontend/src/app/testQueue/useLocalQueue.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import { QueueItemData } from "./TestQueue";
+
+export function useLocalQueue(
+  remoteQueue: QueueItemData[],
+  refreshTime: number
+) {
+  const [localQueue, setLocalQueue] = useState<QueueItemData[]>();
+  const [toRemove, setToRemove] = useState(new Set<string>());
+
+  useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout>;
+    if (!remoteQueue) {
+      return;
+    }
+    // Immediately set local queue if nonexistent
+    if (localQueue === undefined) {
+      setLocalQueue(remoteQueue);
+      return;
+    }
+    // Get discrepancies
+    const newIds = new Set(
+      remoteQueue.map(({ internalId }: QueueItemData) => internalId)
+    );
+    const toRemove = localQueue
+      .filter(({ internalId }) => !newIds.has(internalId))
+      .map(({ internalId }) => internalId);
+    setToRemove(new Set(toRemove));
+
+    // Only requires animation on remove
```

Comment:
Can a new `remoteQueue` have items that were added on some other computer?

Reply:
Yeah, but we don't need to animate them in. They just appear at the bottom of the queue and won't cause a visual shift.